### PR TITLE
Feature/delete and leave topics

### PR DESCRIPTION
**Pull request by tobikuyo**
Users can delete topics that they have created and leave topics that they have joined.

## Before making the Pull Request:

- [x] Is there a description to this pull request that points to a specific user story, feature, or bugfix?
- [x] Are there comments where the code may be unclear?
- [x] Is all dead code, commented out code, etc. removed?
- [ ] Are there tests written for these changes?
- [ ] Does the ReadMe need to be updated?

## Before being merged by the TPL:

- [ ] Has the pull request been reviewed by at least two team members? 
- [ ] Did they leave any comments or suggestions? 
- [ ] Were those suggestions acted on?
- [ ] Are all tests passing?
- [ ] Has the documentation been updated?

### CODE REVIEW

File: labs-ios-starter/Model Controller/UserController+API.swift [hunk 1]
```diff
@@ -419,6 +419,64 @@ extension UserController {
...
+        let requestURL = baseURL
+            .appendingPathComponent("topics")
+            .appendingPathComponent("topic")
+            .appendingPathComponent(topicId.description)
+
+        var request = URLRequest(url: requestURL)
+        request.httpMethod = "DELETE"
+        request.addValue("Bearer \(oktaCredentials.accessToken)", forHTTPHeaderField: "Authorization")
+
+        URLSession.shared.dataTask(with: request) { _, response, error in
```
Comment (hoppel16) [inline labs-ios-starter/Model Controller/UserController+API.swift:434]: We should probably include the "isMock" and use the dataLoader here just to keep it consistent with all the other networking.

File: labs-ios-starter/Model Controller/UserController+API.swift [hunk 2]
```diff
@@ -419,6 +419,64 @@ extension UserController {
...
+        let requestURL = baseURL
+            .appendingPathComponent("topics")
+            .appendingPathComponent("leave")
+            .appendingPathComponent(topicId.description)
+
+        var request = URLRequest(url: requestURL)
+        request.httpMethod = "DELETE"
+        request.addValue("Bearer \(oktaCredentials.accessToken)", forHTTPHeaderField: "Authorization")
+
+        URLSession.shared.dataTask(with: request) { _, response, error in
```
Comment (hoppel16) [inline labs-ios-starter/Model Controller/UserController+API.swift:463]: Same as above

File: labs-ios-starter/Topics/TopicTableViewController.swift [hunk 3]
```diff
@@ -87,4 +97,37 @@ extension TopicTableViewController: UITableViewDataSource, UITableViewDelegate {
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
     }
+
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
```
Comment (hoppel16) [inline labs-ios-starter/Topics/TopicTableViewController.swift:101]: Happy to finally be able to clean up the test topics

**Review by hoppel16 — APPROVED** [review]
Looks good. I still really like how the errors are added as a helper file.

File: labs-ios-starter/Model Controller/UserController+API.swift (hunk 1 above)
Comment (tobikuyo) [reply labs-ios-starter/Model Controller/UserController+API.swift:434]: I didn't include the isMock, because we seem to only have in "GET" requests

Comment (tobikuyo) [reply labs-ios-starter/Model Controller/UserController+API.swift:434]: I didn't include the isMock, because we seem to only have in "GET" requests

**Pull request merged**